src/containers/App.js

```diff
@@ -0,0 +1,98 @@
+// src/containers/App.js
+
+import React, { Component } from 'react'
+
+import { ALL_TODOS, ACTIVE_TODOS, COMPLETED_TODOS } from '../constants';
+import Footer from '../components/Footer';
+import Header from '../components/Header';
+import Todos from '../components/Todos';
+import Todo from '../components/Todo';
+
+class App extends Component {
+	constructor(props) {
+		super(props);
+
+		// JSX supports passing in element arrays
+		/*
+		const array = [
+			<Todo label="First todo..." deleteHandler={this.handleTodoDelete} />,
+			<Todo label="Second todo..." deleteHandler={this.handleTodoDelete} />,
+			<Todo label="Third todo..." deleteHandler={this.handleTodoDelete} />
+		];
+		*/
+		this.filter = this.filter.bind(this);
+		this.handleTodoAdd = this.handleTodoAdd.bind(this);
+		this.handleTodoClear = this.handleTodoClear.bind(this);
+		this.handleTodoDelete = this.handleTodoDelete.bind(this);
+		this.handleTodoComplete = this.handleTodoComplete.bind(this);
+
+		this.state = {
+			lastIndex: 1,
+			show: ALL_TODOS,
+			items: []
+		};		
+	}
+
+	handleTodoAdd(e) {
+		console.log('add item');
+
+		if (e.target.value == '') return 0;
+
+    		if (e.key === 'Enter') {
+			this.setState({ lastIndex : this.state.lastIndex + 1 });
+      			this.setState({ items : this.state.items.concat(<Todo key={this.state.lastIndex} completed={false} label={e.target.value} deleteHandler={this.handleTodoDelete} completeHandler={this.handleTodoComplete}/>)});
+			e.target.value = '';
+    		}
+  	}
+
+	handleTodoDelete(e) {
+		console.log('remove item');
+
+		// this doesn't seem like the right way to do this
+		// but I can't figure out how to get a reference to the todo object that triggered the handler
+		var items = this.state.items.filter(function(todo) { return todo.props.label != e.target.value; });
+		this.setState({ items: items });
+	}
+
+	handleTodoComplete(item)
+	{
+		console.log('item key to complete: ' + item.props.label);
+		
+		var items = this.state.items.filter(function(todo) { 
+			if (todo.props.label == item.props.label) { console.log('found'); /*todo.props.completed = true;*/ }
```

Comment:
Noticed that you have commented out `todo.props.completed = true;`.  I'm guessing that is because you have come across some issues?  We can go through this over the phone.